python/python/tests/test_async_client.py

```diff
@@ -6814,6 +6814,123 @@ async def test_xinfo_groups_xinfo_consumers_edge_cases_and_failures(
         with pytest.raises(RequestError):
             await glide_client.xinfo_consumers(string_key, group_name)
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_xinfo_stream(
+        self, glide_client: TGlideClient, cluster_mode, protocol
+    ):
+        key = get_random_string(10)
+        group_name = get_random_string(10)
+        consumer = get_random_string(10)
+        stream_id0_0 = "0-0"
+        stream_id1_0 = "1-0"
+        stream_id1_1 = "1-1"
+
+        # setup: add stream entry, create consumer group and consumer, read from stream with consumer
+        assert (
+            await glide_client.xadd(
+                key, [("a", "b"), ("c", "d")], StreamAddOptions(stream_id1_0)
+            )
+            == stream_id1_0.encode()
+        )
+        assert await glide_client.xgroup_create(key, group_name, stream_id0_0) == OK
+        assert await glide_client.xreadgroup({key: ">"}, group_name, consumer) == {
+            key.encode(): {stream_id1_0.encode(): [[b"a", b"b"], [b"c", b"d"]]}
+        }
+
+        result = await glide_client.xinfo_stream(key)
+        assert result.get(b"length") == 1
+        expected_first_entry = [stream_id1_0.encode(), [b"a", b"b", b"c", b"d"]]
+        assert result.get(b"first-entry") == expected_first_entry
+
+        # only one entry exists, so first and last entry should be the same
+        assert result.get(b"last-entry") == expected_first_entry
+
+        # call XINFO STREAM with a byte string arg
+        result2 = await glide_client.xinfo_stream(key.encode())
+        assert result2 == result
+
+        # add one more entry
+        assert (
+            await glide_client.xadd(
+                key, [("foo", "bar")], StreamAddOptions(stream_id1_1)
+            )
+            == stream_id1_1.encode()
+        )
+
+        result_full = await glide_client.xinfo_stream_full(key, count=1)
+        print(result_full)
```

Comment:
nit: remove prints once you finish debugging

Reply:
forgot to remove, thanks